common/src/main/java/org/figuramc/figura/mixin/render/renderers/SkullBlockRendererMixin.java

```diff
@@ -89,11 +88,25 @@ public void render(SkullBlockEntity skullBlockEntity, float f, PoseStack poseSta
 
     @Override
     public boolean shouldRenderOffScreen(SkullBlockEntity blockEntity) {
-        return avatar == null ? BlockEntityRenderer.super.shouldRenderOffScreen(blockEntity) : avatar.permissions.get(Permissions.OFFSCREEN_RENDERING) == 1;
+    	// if we have an avatar, check permissions
+    	if(avatar != null  && avatar.permissions != null) {
+    		// if we have permissions to render, then we render, else we don't render.
+    		if(avatar.permissions.get(Permissions.OFFSCREEN_RENDERING) == 1) {
+    		    return true;
+    		}
+    		return false;
+    	}
+    	//no avatar present, default rendering used.  
+    	return BlockEntityRenderer.super.shouldRenderOffScreen(blockEntity);
```

Comment:
Again this whole thing can be simplified to
```suggestion
    	return avatar == null || avatar.permissions == null ? BlockEntityRenderer.super.shouldRenderOffScreen(blockEntity) : avatar.permissions.get(Permissions.OFFSCREEN_RENDERING) == 1;
```
Use ternaries to your advantage. Also ```avatar.permissions.get(Permissions.OFFSCREEN_RENDERING) == 1``` is already a boolean, there is no reason to expand the logic there either. Something tells me you really should review the basics, no offense meant, but we like to keep things simple and compact. Also the extra comments aren't necessary or wanted when the code is already self explanatory.

Reply:
I guess code tends to expand during debugging, I'll remember that.